test/util/isDomainSpecifiedByUser.spec.ts

```diff
@@ -85,3 +89,186 @@ describe('isDomainSpecifiedByUser', () => {
     expect(isDomainSpecifiedByUser(undefined as never, true, 'number')).toBe(false);
   });
 });
+
+describe('parsing axis domain provided by user', () => {
+  // These are valid domains without any further questions
+  const validCases: MyTestCases = [
+    { domain: [-100, 100], expected: [-100, 100] },
+    { domain: [() => 1, () => 2], expected: [1, 2] },
```

Comment:
I think we're missing a case for 

([dataMin, dataMax]) => [dataMin, dataMax]

I see individual functions for each but there's also a version where the whole thing is a function and both get passed to it

Reply:
Here is a test case for dataMin, dataMax: https://github.com/PavelVanecek/recharts/blob/9c7a275b59887ac2c649228ef526169d22bade16/test/util/isDomainSpecifiedByUser.spec.ts#L144

Here is a test case for one function that accepts an array: https://github.com/PavelVanecek/recharts/blob/9c7a275b59887ac2c649228ef526169d22bade16/test/util/isDomainSpecifiedByUser.spec.ts#L147

The tests are split because by using `dataMin` and `dataMax` it forces the selector to iterate all data. This is mentioned in code as a performance optimization: https://github.com/recharts/recharts/blob/6e9f9fe5b06ab939135a48f4c68e124d89e3f337/src/chart/generateCategoricalChart.tsx#L360

Now, I have my doubts on the effectiveness of such optimization (since it has to iterate the data _anyway_, the chart cannot render without iterating it all) but again ... let's keep it as-is until we find a good reason to change it.

Reply:
When domain is a function it gives the array of dataMin dataMax as a parameter to that function. Will send an example in a sec

Reply:
Does this case cover it?

Reply:
Kind of but not really since `d` isn't data but rather an array of `[dataMin, dataMax]` so you could just make that case `([dMin]) => [dMin, 999]`

and then would want another case for 
`([dMin, dMax]) => [dMin, dMax + 100]` or something like that

Reply:
I can add that

Reply:
Sweet looks good thanks